caddyhttp/markdown/setup.go

```diff
@@ -75,6 +75,14 @@ func markdownParse(c *caddy.Controller) ([]*Config, error) {
 			md.Extensions[".mdown"] = struct{}{}
 		}
 
+		// Make a list of index files to match extensions
+		i := 0
+		md.IndexFiles = make([]string, len(md.Extensions))
+		for ext := range md.Extensions {
+			indexName := "index" + ext
+			md.IndexFiles[i] = indexName
+			i++
+		}
```

Comment:
I think this whole change can be simplified:

```go
for ext := range md.Extensions {
    md.IndexFiles = append(md.IndexFiles, "index" + ext)
}
```

Pre-allocating is good for a slice that will be very large, but when there's like 5 or 10 things, I prefer shorter, less error-prone code.